pkg/minikube/cluster/cluster.go

```diff
@@ -300,9 +303,53 @@ func createVirtualboxHost(config MachineConfig) drivers.Driver {
 	return d
 }
 
+func isISONotCachedInMinikubeDir(isoFilePath string) bool {
+	if _, err := os.Stat(isoFilePath); os.IsNotExist(err) {
+		return true
+	}
+	return false
+}
+
+func getISOFilePathFromConfig(config MachineConfig) string {
+	split := strings.Split(config.MinikubeISO, "/")
+	isoFilename := split[len(split)-1]
+	return filepath.Join(constants.Minipath, "iso", isoFilename)
+}
+
+func modifyConfigForCachedIso(isoFilePath string, config *MachineConfig) error {
```

Comment:
I'd find this easier to understand if this were a method on MinikubeConfig, and it only did the translation. Something like:

`(m *MinikubeConfig) GetISOCacheLocation() string`

Then you could change all the current callers (createVirtualboxHost, create*Host) of config.MinikubeISO to call that function instead of just accessing the property.

Then, in `createHost` you should do the actual caching before the drivers get created.